README.md

```diff
@@ -1,16 +1,659 @@
+  nnc String  -- Nonce
+  ext nullable Any (implicit null) -- Extended fields
+}
+
+type SignedInvocation struct {
+  inv Invocation (rename "ucan/invoke") 
+  sig VarSig  -- Signature
+}
+
+type VarSig Bytes
+
+type Scope enum {
+  | All ("*")
+  | Action
+}
+
+type Action enum {
+  using  URI 
+  do     Ability
+  inputs Any
+} 
+```
+
+## 3.3 JSON Examples
+
+### 3.3.1 Run All
+
+ ``` js
+{
+  "ucan/invoke": {
+    "v": "0.1.0",
+    "prf": ["bafkreifzjut3te2nhyekklss27nh3k72ysco7y32koao5eei66wof36n5e"],
+    "nnc": "abcdef",
+    "run": "*", // Explicitly "run all"
+    "ext": null
+  },
+  "sig": "bdNVZn_uTrQ8bgq5LocO2y3gqIyuEtvYWRUH9YT-SRK6v_SX8bjt-VZ9JIPVTdxkWb6nhVKBt6JGpgnjABpOCA"
+}
+```
+
+### 3.3.2 Promise Pipelines
+
+Promise pipelines are handled in more detail in [section 5](#5-promise-pipelining). In brief, they enable taking the output of an action that has not yet run as the input to another action. Here is a simple example:
+
+``` js
+{
+  "ucan/invoke": {
+    "v": "0.1.0",
+    "prf": ["bafkreifzjut3te2nhyekklss27nh3k72ysco7y32koao5eei66wof36n5e"],
+    "nnc": "02468",
+    "ext": null,
+    "run": {
+      "notify-bob": {
+        "using": "mailto://alice@example.com",
+        "do": "msg/send",
+        "inputs": [
+          {
+            "to": "bob@example.com",
+            "subject": "DNSLink for example.com",
+            "body": "Hello Bob!"
+          }
+        ]
+      },
+      "log-as-done": {
+        "using": "https://example.com/report"
+        "do": "crud/update"
+        "inputs": {
+          "from": "mailto://alice@exmaple.com",
+          "to": ["bob@exmaple.com"],
+          "event": "email-notification",
+          "value": {"ucan/promise": ["/", "notify-bob"]} // Pipelined promise
+        }
+      }
+    }
+  },
+  "sig": "5vNn4--uTeGk_vayyPuNTYJ71Yr2nWkc6AkTv1QPWSgetpsu8SHegWoDakPVTdxkWb6nhVKAz6JdpgnjABppC7"
+}
+```
+
+# 4 Receipt
+
+An invocation receipt is an attestation of the output of an invocation. A receipt MUST be signed by the executor (the `aud` of the associated UCAN).
+
+Note that this does not guarantee correctness of the result! The statement's veracity MUST be only understood as an attestation from the executor.
+
+## 4.1 Fields
+
+| Field | Type          | Description                                                                                      | Required | Default |
+|-------|---------------|--------------------------------------------------------------------------------------------------|----------|---------|
+| `inv` | `&Invocation` | CID of the Invocation that generated this response                                               | Yes      |         |
+| `rlt` | `{CID: Any}`  | The results of each call, indexed by the CID of the `dag-cbor` encoded [Action](#32-ipld-schema) | Yes      |         |
+| `v`   | `SemVer`      | SemVer of the UCAN invocation object schema                                                      | Yes      |         |
```

Comment:
Is this really necessary, we already have a link to `inv` containing it, this just creates an opportunity for two to be misaligned.

Reply:
And if this is indeed required any reason it needs to be required ?

Reply:
Okay, that's a good point! It MUST be the same Receipt version as what it contains 👍